action.yml

```diff
@@ -1,29 +1,37 @@
 # See: https://docs.github.com/en/actions/creating-actions
 name: Gem Push
 author: FreeAgent
-description: Push gem packages to a rubygems compatible repo
+description: Push gem packages to a rubygems compatible repository
 inputs:
   package-glob:
-    description: "File glob to match the .gem files to push"
+    description: File glob to match the .gem files to push
     default: "pkg/*.gem"
+  release:
+    description: Whether to push release versions
+    default: true
+  pre-release:
+    description: Whether to push pre-release versions
+    default: true
+  tag-release:
+    description: After pushing a new gem version, git tag with the version string
+    default: true
+outputs:
+  pushed-version:
+    description: "The version of the gem pushed to the repository"
+    value: ${{ steps.push-gem.outputs.pushed-version }}
 
 runs:
   using: "composite"
   steps:
     - name: Push Gem
+      id: push-gem
       shell: bash
       env:
         # Expects GEM_HOST and GEM_HOST_API_KEY to be set
-        GEM_GLOB: ${{ inputs.package-glob }}
+        INPUT_PACKAGE_GLOB: ${{ inputs.package-glob }}
```

Comment:
I feel like passing args with env vars is a bit indirect, the script has opt parsing, why not pass these as command-line args?

Reply:
The ruby script has arg passing, but the bash script doesn't. I could write a bash option parser but that is more work than doing a ruby one. This is a pattern I picked up from digging in other actions.

Reply:
ah, the action calls the bash script which then calls the ruby script, ok. I feel like inlining the bash script into the action avoids passing arguments then.

I think you separated them to make testing easier? For workflow testing, I think people like [`act`](https://github.com/nektos/act)

Reply:
act doesn't support composite actions. Partly testing but also linting. In a seperate file I can just open and have all my local setup shell linting running. Otherwise I have to keep copy and pasting. Also syntax highlighting.

Reply:
I see `act` will support composite commands soon (https://github.com/nektos/act/pull/608). I reckon running tests on a workflow is better than on an action, because you can verify that your example workflows work as they should.

But that's enough questions, this makes sense now.